CHANGELOG.md

```diff
@@ -1,3 +1,137 @@
+- Fixed ICE when using unions of metaclasses. ([#6307], thanks @asterite)
+- Fixed ICE related to literal type guessing and generic types hierarchy. ([#6341], thanks @asterite)
+- Fixed ICE related to `not` and inlinable values. ([#6452], thanks @asterite)
+- Fixed rebind variables type in while condition after analyzing its body. ([#6295], thanks @asterite)
+- Fixed corner cases regarding automatic casts and method instantiation. ([#6284], thanks @asterite)
+- Fixed parsing of `\A` (and others) inside `%r{...}` inside macros. ([#6282], thanks @asterite)
+- Fixed parsing of of named tuple inside generic type arguments. ([#6413], thanks @asterite)
+- Fixed disallow cast from module class to virtual metaclass. ([#6320], thanks @asterite)
+- Fixed disallow `return` inside a constant's value. ([#6347], thanks @asterite)
+- Fixed debug info for closured self. ([#6346], thanks @asterite)
+- Fixed parsing error of newline before closing macro. ([#6382], thanks @asterite)
+- Fixed missing error if constant has `NoReturn` type. ([#6411], thanks @asterite)
+- Fixed give proper error when doing sizeof uninstantiated generic type. ([#6418], thanks @asterite)
+- Fixed private aliases at top-level are now considered private. ([#6432], thanks @asterite)
+- Refactor codegen of binary operators. ([#6330], thanks @bcardiff)
+- Refactor use `JSON::Serializable` instead of `JSON.mapping`. ([#6308], thanks @kostya)
+- Change how metaclasses are shown. Use `Foo.class` instead of `Foo:Class`. ([#6439], thanks @RX14)
+
+## Tools
+
+- Flatten project structure created by `crystal init`. ([#6317], thanks @straight-shoota)
+
+### Formatter
+
+- Fixed formatting of `{ {1}.foo, ...}` like expressions. ([#6300], thanks @asterite)
+- Fixed formatting of `when` with numbers. Use right alignment only if all are number literals. ([#6392], thanks @MakeNowJust)
+- Fixed formatting of comment in case's else. ([#6393], thanks @MakeNowJust)
+- Fixed code fence when language is not crystal will not be formatted. ([#6424], thanks @asterite)
+
+### Doc generator
+
+- Add line numbers at link when there are duplicated filenames in "Defined in:" section. ([#6280], thanks @r00ster91)
+- Fix docs navigator not scrolling into open type on page load. ([#6420], thanks @soanvig)
+
+### Playground
+
+## Others
+
+- Fixed `system_spec` does no longer emit errors messages on BSD platforms. ([#6289], thanks @jcs)
+- Fixed compilation issue when running spec against compiler and std together. ([#6312], thanks @straight-shoota)
+- Add support for LLVM 6.0. ([#6381], [#6380], [#6383], thanks @felixbuenemann)
+- CI improvements and housekeeping. ([#6313], [#6337], [#6407], [#6408], [#6315], thanks @bcardiff, @MakeNowJust, @r00ster91, @maiha)
+- Refactor platform specifics from `ENV` to `Crystal::System::Env` and implement for windows. ([#6333],
+- Add an optional argument to `String#check_no_null_byte.thanks @straight-shoota`)
```

Comment:
This change is already listed in *Text*, but the one above misses the attribution.